### ISSUE TITLE
[youtube:tab] "Incomplete data received" on "Redownloading playlist API JSON with unavailable videos"

### ISSUE THREAD

**Issue by Crystal-RainSlide** [yt-dlp/yt-dlp#8283]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.09.24** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

US

### Provide a description that is worded well enough to be understood

When listing (`--flat-playlist`) or downloading a YT playlist, a series of warnings and an error about `Incomplete data received` will appear, before starting download the videos normally.

Also, the count of the video items seems wrong, it will say:
``` log
[youtube:tab] Playlist {playlist name} Downloading n+1 items of n
[download] Downloading item 1 of n+1
...
[download] Downloading item n of n+1
```

Instead of:
``` log
[youtube:tab] Playlist {playlist name} Downloading n items of n
[download] Downloading item 1 of n
...
[download] Downloading item n of n
```

- I can still download all the available playlist videos without problem.
- I can confirm this still happens when `--compat-options no-youtube-unavailable-videos` is added.
Something is wrong in a call to the `_extract_response()` function without the `note` func arg.

### Related code

> `Redownloading playlist API JSON with unavailable videos`:
> https://github.com/yt-dlp/yt-dlp/blob/b095fd3fa9d58a65dc9b830bd63b9d909422aa86/yt_dlp/extractor/youtube.py#L5262

> `Incomplete data received`:
> https://github.com/yt-dlp/yt-dlp/blob/b095fd3fa9d58a65dc9b830bd63b9d909422aa86/yt_dlp/extractor/youtube.py#L997

> `[youtube:tab] Playlist {playlist name} Downloading n+1 items of n`:
> https://github.com/yt-dlp/yt-dlp/blob/b095fd3fa9d58a65dc9b830bd63b9d909422aa86/yt_dlp/YoutubeDL.py#L1960

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--flat-playlist', 'https://www.youtube.com/playlist?list=PL4F761764995C5F15']
[debug] Portable config "D:\Programs\Media\yt-dlp\yt-dlp.conf": ['--proxy', 'http://127.0.0.1:7890/', '--downloader', 'aria2c', '-N', '4', '-r', '128M', '--progress', '--console-title']
[debug] Encodings: locale cp936, fs utf-8, pref cp936, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.09.24 [088add956] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109420-g262e7439c6-20221214 (setts), ffprobe N-109420-g262e7439c6-20221214, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2023.07.22, mutagen-1.47.0, sqlite3-3.35.5, websockets-11.0.3
[debug] Proxy map: {'all': 'http://127.0.0.1:7890/'}
[debug] Loaded 1886 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.09.24, Current version: stable@2023.09.24
Current Build Hash: 125465ac4416bcd4bf1632b899630ceecc08b341e83fdf10bd34700c58ea5152
yt-dlp is up to date (stable@2023.09.24)
[youtube:tab] Extracting URL: https://www.youtube.com/playlist?list=PL4F761764995C5F15
[youtube:tab] PL4F761764995C5F15: Downloading webpage
[youtube:tab] PL4F761764995C5F15: Redownloading playlist API JSON with unavailable videos
[download] Downloading playlist: Madness Combat Series
[youtube:tab] PL4F761764995C5F15 page 1: Downloading API JSON
WARNING: [youtube:tab] Incomplete data received. Retrying (1/3)...
[youtube:tab] PL4F761764995C5F15 page 1: Downloading API JSON
WARNING: [youtube:tab] Incomplete data received. Retrying (2/3)...
[youtube:tab] PL4F761764995C5F15 page 1: Downloading API JSON
WARNING: [youtube:tab] Incomplete data received. Retrying (3/3)...
[youtube:tab] PL4F761764995C5F15 page 1: Downloading API JSON
ERROR: Incomplete data received
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1567, in wrapper
  File "yt_dlp\utils\_utils.py", line 2427, in <lambda>
  File "yt_dlp\utils\_utils.py", line 2195, in __getitem__
  File "yt_dlp\extractor\youtube.py", line 4933, in _entries
  File "yt_dlp\extractor\youtube.py", line 944, in _extract_response
  File "yt_dlp\utils\_utils.py", line 5112, in __iter__
  File "yt_dlp\extractor\common.py", line 3760, in _error_or_warning
  File "yt_dlp\utils\_utils.py", line 5120, in report_retry
yt_dlp.utils.ExtractorError: Incomplete data received

[youtube:tab] Playlist Madness Combat Series: Downloading 18 items of 17
[debug] The information of all playlist entries will be held in memory
[download] Downloading item 1 of 18
[download] Downloading item 2 of 18
[download] Downloading item 3 of 18
[download] Downloading item 4 of 18
[download] Downloading item 5 of 18
[download] Downloading item 6 of 18
[download] Downloading item 7 of 18
[download] Downloading item 8 of 18
[download] Downloading item 9 of 18
[download] Downloading item 10 of 18
[download] Downloading item 11 of 18
[download] Downloading item 12 of 18
[download] Downloading item 13 of 18
[download] Downloading item 14 of 18
[download] Downloading item 15 of 18
[download] Downloading item 16 of 18
[download] Downloading item 17 of 18
[download] Finished downloading playlist: Madness Combat Series
```

**Comment by Crystal-RainSlide** [comment]
Maybe also see: https://github.com/yt-dlp/yt-dlp/issues/6289

**Comment by coletdjnz** [comment]
fixed in master/nightly

**Comment by dirkf** [comment]
This is causing test YoutubePlaylist_2 to fail, even after the fix for #8206.

The test needs to ignore the retry warnings and the former "Giving up" error that (I gather) is now a warning:
```diff
--- old/yt_dlp/extractor/youtube.py
+++ new/yt_dlp/extractor/youtube.py
@@ -6687,7 +6687,7 @@ class YoutubePlaylistIE(InfoExtractor):
             'uploader_url': 'https://www.youtube.com/@milan5503',
             'availability': 'public',
         },
-        'expected_warnings': [r'[Uu]navailable videos? (is|are|will be) hidden'],
+        'expected_warnings': [r'[Uu]navailable videos? (is|are|will be) hidden', 'Retrying', 'Giving up'],
     }, {
         'url': 'http://www.youtube.com/embed/_xDOZElKyNU?list=PLsyOSbh5bs16vubvKePAQ1x3PhKavfBIl',
         'playlist_mincount': 455,
```


Below log shows today's master, but similar behaviour occurs with 2023.06.22 once the consent issue is fixed (it also reported "Downloading 5 of 4 items" but this isn't happening now).

The log shows the failing test and than fetching the test URL directly.

Presumably the fix for 

As CI download tests are all being skipped, I couldn't find examples in the workflow log.

yt-dl doesn't (yet) know how to `_reload_with_unavailable_videos()`, which is being tested here, and now returns an empty playlist.

$ python3.9 test/test_download.py TestDownload.test_YoutubePlaylist_2
[debug] Loaded 1890 extractors
[youtube:playlist] Extracting URL: https://www.youtube.com/embed/videoseries?list=PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu
[youtube:tab] Extracting URL: https://www.youtube.com/playlist?list=PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu: Downloading webpage
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu: Redownloading playlist API JSON with unavailable videos
[download] Downloading playlist: JODA15
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu page 1: Downloading API JSON
ERROR: [youtube:tab] Incomplete data received. Retrying (1/3)...; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1567, in wrapper
    return func(self, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 2427, in <lambda>
    return type(self.ydl)._handle_extraction_exceptions(lambda _, i: self._entries[i])(self.ydl, i)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 2195, in __getitem__
    self._cache.extend(itertools.islice(self._iterable, n))
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/youtube.py", line 4955, in _entries
    response = self._extract_response(
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/youtube.py", line 1001, in _extract_response
    should_retry = next(icd_retries, None)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 5114, in __iter__
    self.error_callback(self.error, self.attempt, self.retries)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/common.py", line 3760, in _error_or_warning
    RetryManager.report_retry(
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 5128, in report_retry
    warn(f'{e}. Retrying{format_field(suffix, None, " %s")} ({count}/{retries})...')
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/common.py", line 1132, in report_warning
    self._downloader.report_warning(msg, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/test/helper.py", line 315, in _report_warning
    real_warning(w, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/test/test_download.py", line 50, in report_warning
    raise ExtractorError(message)
yt_dlp.utils.ExtractorError: [youtube:tab] Incomplete data received. Retrying (1/3)...; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

E
======================================================================
ERROR: test_YoutubePlaylist_2 (__main__.TestDownload):
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1567, in wrapper
    return func(self, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 2427, in <lambda>
    return type(self.ydl)._handle_extraction_exceptions(lambda _, i: self._entries[i])(self.ydl, i)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 2195, in __getitem__
    self._cache.extend(itertools.islice(self._iterable, n))
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/youtube.py", line 4955, in _entries
    response = self._extract_response(
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/youtube.py", line 1001, in _extract_response
    should_retry = next(icd_retries, None)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 5114, in __iter__
    self.error_callback(self.error, self.attempt, self.retries)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/common.py", line 3760, in _error_or_warning
    RetryManager.report_retry(
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 5128, in report_retry
    warn(f'{e}. Retrying{format_field(suffix, None, " %s")} ({count}/{retries})...')
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/extractor/common.py", line 1132, in report_warning
    self._downloader.report_warning(msg, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/test/helper.py", line 315, in _report_warning
    real_warning(w, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/test/test_download.py", line 50, in report_warning
    raise ExtractorError(message)
yt_dlp.utils.ExtractorError: [youtube:tab] Incomplete data received. Retrying (1/3)...; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/df/Documents/src/yt-dlp-new/test/test_download.py", line 172, in test_template
    res_dict = ydl.extract_info(
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1556, in extract_info
    return self.__extract_info(url, self.get_info_extractor(key), download, extra_info, process)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1567, in wrapper
    return func(self, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1723, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1802, in process_ie_result
    return self.extract_info(
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1556, in extract_info
    return self.__extract_info(url, self.get_info_extractor(key), download, extra_info, process)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1567, in wrapper
    return func(self, *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1723, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1852, in process_ie_result
    return self.__process_playlist(ie_result, download)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1927, in __process_playlist
    entries = resolved_entries = list(entries)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 738, in _iter
    for x in iterable:
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 2392, in get_requested_items
    for i, entry in self[index]:
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 2454, in __getitem__
    entry = self._getter(i)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/utils/_utils.py", line 2427, in get_entry
    return type(self.ydl)._handle_extraction_exceptions(lambda _, i: self._entries[i])(self.ydl, i)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1585, in wrapper
    self.report_error(str(e), e.format_traceback())
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 1045, in report_error
    self.trouble(f'{self._format_err("ERROR:", self.Styles.ERROR)} {message}', *args, **kwargs)
  File "/home/df/Documents/src/yt-dlp-new/yt_dlp/YoutubeDL.py", line 984, in trouble
    raise DownloadError(message, exc_info)
yt_dlp.utils.DownloadError: ERROR: [youtube:tab] Incomplete data received. Retrying (1/3)...; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

----------------------------------------------------------------------
Ran 1 test in 6.050s

FAILED (errors=1)
$ # Fetching the same playlist directly
$ python3.9 -m yt_dlp -v --flat-playlist 'https://www.youtube.com/embed/videoseries?list=PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu'
[debug] Command-line config: ['-v', '--flat-playlist', 'https://www.youtube.com/embed/videoseries?list=PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.10.13 [b634ba742] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 4a601c9ef
[debug] Python 3.9.18 (CPython i686 32bit) - Linux-4.4.0-210-generic-i686-with-glibc2.23 (OpenSSL 1.1.1w  11 Sep 2023, glibc 2.23)
[debug] exe versions: ffmpeg 4.3, ffprobe 4.3
[debug] Optional libraries: Cryptodome-3.11.0, certifi-2019.11.28, requests-2.25.1, secretstorage-3.2.0, sqlite3-3.37.2, urllib3-1.25.11
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1890 extractors
[youtube:playlist] Extracting URL: https://www.youtube.com/embed/videoseries?list=PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu
[youtube:tab] Extracting URL: https://www.youtube.com/playlist?list=PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu: Downloading webpage
WARNING: [youtube:tab] YouTube said: INFO - 4 unavailable videos are hidden
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu: Redownloading playlist API JSON with unavailable videos
[download] Downloading playlist: JODA15
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu page 1: Downloading API JSON
WARNING: [youtube:tab] Incomplete data received. Retrying (1/3)...
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu page 1: Downloading API JSON
WARNING: [youtube:tab] Incomplete data received. Retrying (2/3)...
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu page 1: Downloading API JSON
WARNING: [youtube:tab] Incomplete data received. Retrying (3/3)...
[youtube:tab] PL6IaIsEjSbf96XFRuNccS_RuEXwNdsoEu page 1: Downloading API JSON
WARNING: [youtube:tab] Incomplete data received. Giving up after 3 retries
[youtube:tab] Playlist JODA15: Downloading 4 items of 4
[debug] The information of all playlist entries will be held in memory
[download] Downloading item 1 of 4
[download] Downloading item 2 of 4
[download] Downloading item 3 of 4
[download] Downloading item 4 of 4
[download] Finished downloading playlist: JODA15
$
